openmc/checkvalue.py

```diff
@@ -0,0 +1,337 @@
+import copy
+from collections.abc import Iterable
+
+import numpy as np
+
+
+def check_type(name, value, expected_type, expected_iter_type=None):
+    """Ensure that an object is of an expected type. Optionally, if the object is
+    iterable, check that each element is of a particular type.
+
+    Parameters
+    ----------
+    name : str
+        Description of value being checked
+    value : object
+        Object to check type of
+    expected_type : type or Iterable of type
+        type to check object against
+    expected_iter_type : type or Iterable of type or None, optional
+        Expected type of each element in value, assuming it is iterable. If
+        None, no check will be performed.
+
+    """
+
+    if not isinstance(value, expected_type):
+        if isinstance(expected_type, Iterable):
+            msg = 'Unable to set "{0}" to "{1}" which is not one of the ' \
+                  'following types: "{2}"'.format(name, value, ', '.join(
+                      [t.__name__ for t in expected_type]))
+        else:
+            msg = 'Unable to set "{0}" to "{1}" which is not of type "{2}"'.format(
+                name, value, expected_type.__name__)
+        raise TypeError(msg)
+
+    if expected_iter_type:
+        if isinstance(value, np.ndarray):
+            if not issubclass(value.dtype.type, expected_iter_type):
+                msg = 'Unable to set "{0}" to "{1}" since each item must be ' \
```

Comment:
This assignment of message is never used. The variable is always reset before use in subsequent code.